app/assets/javascripts/discourse/app/components/group-imap-email-settings.js

```diff
@@ -0,0 +1,112 @@
+import Component from "@ember/component";
+import { later } from "@ember/runloop";
+import I18n from "I18n";
+import bootbox from "bootbox";
+import { isEmpty } from "@ember/utils";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import discourseComputed, { on } from "discourse-common/utils/decorators";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+
+export default Component.extend({
+  tagName: "",
+  group: null,
+  form: null,
+
+  @discourseComputed("group.imap_mailboxes")
+  mailboxes(imapMailboxes) {
+    if (!imapMailboxes) {
+      return [];
+    }
+    return imapMailboxes.map((mailbox) => ({ name: mailbox, value: mailbox }));
+  },
+
+  @discourseComputed("group.imap_mailbox_name", "mailboxes.length")
+  mailboxSelected(mailboxName, mailboxesSize) {
+    if (mailboxesSize === 0) {
+      return true;
+    }
+
+    return !isEmpty(mailboxName);
+  },
+
+  @action
+  resetSettingsValid() {
+    if (this.initializing) {
+      return;
+    }
+    this.set("group.imapSettingsValid", false);
+  },
+
+  @on("init")
+  _fillForm() {
+    this.initializing = true;
```

Comment:
Because I have the `resetSettingsValid` method that is fired on change of any of the fields:

```
        {{input type="text" name="imap_port" value=form.imap_port tabindex="4" onChange=(action "resetSettingsValid")}}
```

I am setting the form fields here, and so `resetSettingsValid` is triggered, and then ember throws an error because the `"group.imapSettingsValid"` field is changed twice in the same run/render loop. If there is an alternative I will change it gladly, it feels hacky to me as well.

Reply:
Actually nevermind with this, now that I moved imapSettingsValid off the group this is no longer needed :+1: